### PR TITLE
build: add deployment script

### LINKED ISSUE #330: Add deployment script
### Expected Behavior
run a script installing all OpenMLDB components(zookeeper, nameserver and tablet).  Deployment document reference [here](https://github.com/4paradigm/OpenMLDB/blob/main/docs/cn/deploy.md)
```
sh tools/install.sh
```

### Notice
* report an error and exit if java is not installed
* current port may be used by other process

### PR DESCRIPTION
**Pull request by ProProGG**
* **Please check if the PR fulfills these requirements**

- The commit message follows our guidelines
- Tests for the changes have been added (for bug fixes / features)
- Docs have been added / updated (for bug fixes / features)

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
feature

* **What is the current behavior?** (You can also link to an open issue here)
https://gitee.com/paradigm4/OpenMLDB/issues/I48BZK
https://github.com/4paradigm/OpenMLDB/issues/330

* **What is the new behavior (if this is a feature change)?**
a script to install all OpenMLDB components

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)

* **Other information**:

### CODE REVIEW

File: tools/install.sh [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    mv openmldb-0.2.2-linux openmldb-ns-0.2.2
+    echo 'install tablet'
+    tar -zxvf openmldb-0.2.2-linux.tar.gz
+    mv openmldb-0.2.2-linux openmldb-tablet-2.2.0
+  else
+    echo 'download fail ; try script again'
+    exit 1
+  fi
+fi
+
```
Comment (dl239) [inline tools/install.sh:43]: need to start zk ns and tablet. see deployment [doc](https://github.com/4paradigm/OpenMLDB/blob/main/docs/cn/deploy.md) and notice in  (https://github.com/4paradigm/OpenMLDB/issues/330)

File: tools/install.sh [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  exit 1
+fi
+
+if ! [ -x "$(command -v wget)" ]; then
+  echo 'Error: wget is not installed.' >&2
+  exit 1
+fi
+
+echo 'install zookeeper'
+FILE=zookeeper-3.4.14.tar.gz
```
Comment (dl239) [inline tools/install.sh:14]: `ZK=zookeeper-3.4.14`
`wget  "https://archive.apache.org/dist/zookeeper/${ZK}/${ZK}.tar.gz"`

File: tools/install.sh (hunk 1 above)
Comment (ProProGG) [reply tools/install.sh:43]: As zk's ip and port can be set by users, it can be difficult to start  ns and tablet in the script.

Comment (dl239) [reply tools/install.sh:43]: hi heiyan1shengdun,  some advices as follows:
*  exit and print warnings if zk port is unavailable
*  try to change another port if current ns or tablet port is unavailable

Comment (ProProGG) [reply tools/install.sh:43]: What confuses me is whether zk is running on a local PC or on  some servers. Therefore , i can not get ips , which is neccessary for ns and tablet.

Comment (dl239) [reply tools/install.sh:43]: Deployment script will install zk at first.  use `hostname -i` get local ip

File: tools/install.sh [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+sed -i "s:--enable_distsql=[a-zA-Z0-9//.:]*:--enable_distsql=true:" conf/nameserver.flags
+sh bin/start.sh start nameserver
+cd ..
+echo 'nameserver Ok'
+ 
+echo 'starting tabnet'
+cd openmldb-tablet-2.2.0
+IP=$(hostname -i)
+#get available port
+ns_port=9527
```
Comment (dl239) [inline tools/install.sh]: tablet_port

File: tools/install.sh [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+fi
+#modify tabnet config
+sed -i "s:--endpoint=[a-zA-Z0-9//.:]*:--endpoint=${IP}\:${ns_port}:" conf/tablet.flags
+sed -i "s:--role=[a-zA-Z0-9//.:]*:--role=tablet:" conf/tablet.flags
+sed -i "s:--zk_cluster=[a-zA-Z0-9//.:]*:--zk_cluster=${IP}\:${clientPort}:" conf/tablet.flags
+sed -i "s:--zk_root_path=[a-zA-Z0-9//.:]*:--zk_root_path=/openmldb_cluster:" conf/tablet.flags
+sed -i "s:--enable_distsql=[a-zA-Z0-9//.:]*:--enable_distsql=true:" conf/tablet.flags
+sh bin/start.sh start tablet
+cd ..
+echo 'tabnet Ok'
```
Comment (dl239) [inline tools/install.sh]: tablet

File: tools/install.sh [hunk 5]
```diff
@@ -0,0 +1,127 @@
+#!/bin/bash
```
Comment (vagetablechicken) [inline tools/install.sh:1]: you can `set -e` here, to avoid SC2164 warning.

File: tools/install.sh [hunk 6]
```diff
@@ -0,0 +1,127 @@
...
+  else
+    echo 'download fail ; try script again'
+    exit 1
+  fi
+fi
+
+echo 'install complete'
+
+echo 'starting zookeeper'
+cd zookeeper-3.4.14
```
Comment (vagetablechicken) [inline tools/install.sh]: so many `zookeeper-3.4.14`, it should be a variable?

File: tools/install.sh (hunk 5 above)
Comment (ProProGG) [reply tools/install.sh:1]: Thank you for your suggestions!

**Comment by dl239** [comment]
@heiyan1shengdun hi，could you fix the comments above, thank you

**Review by dl239 — APPROVED** [review]

**Review by vagetablechicken — APPROVED** [review]

**Pull request merged**